_posts/2019-08-13-how-to-deploy-production-grade-vpc-aws.adoc

```diff
@@ -970,22 +990,25 @@ https://www.terraform.io/docs/providers/terraform/d/remote_state.html[terraform_
 data "terraform_remote_state" "mgmt_vpc" {
   backend = "s3"
 
-  config {
+  config = {
     region = var.terraform_state_aws_region
     bucket = var.terraform_state_s3_bucket
-    key    = "${var.aws_region}/mgmt/vpc/terraform.tfstate"
+    # TODO: the key didn't work for me. this is what I had to use.
+    # Based on the folder structure we recommend in [configure_wrapper_module] (management)
+    key = "prod/us-west-1/prod/networking/vpc-mgmt/terraform.tfstate"
+
   }
```

Comment:
We actually should update this to use [the terragrunt `dependency` pattern](https://terragrunt.gruntwork.io/docs/features/execute-terraform-commands-on-multiple-modules-at-once/#passing-outputs-between-modules), which is a more robust solution.

Reply:
What I ended up doing is _slightly_ different to this line-- I created env.hcl and pulled in locals from those parent hcl files, so this ended up being a string interpolated with vars.

But! About the dependency pattern-- ok, I see. Will have to make a note to update this.

Reply:
- [ ] Use [the terragrunt `dependency` pattern](https://terragrunt.gruntwork.io/docs/features/execute-terraform-commands-on-multiple-modules-at-once/#passing-outputs-between-modules)

Reply:
Is this still worth doing now?